packages/synapse-react-client/src/components/Plot/PlotsContainer.tsx

```diff
@@ -0,0 +1,323 @@
+    setPlotUiStateArray(plotUiStateArray => {
+      return plotUiStateArray.map((item, index) => {
+        if (shouldShowMore) {
+          // show everything
+          return { ...item, isHidden: false }
+        }
+        // otherwise hide everything except the first three items
+        return { ...item, isHidden: index >= DEFAULT_VISIBLE_PLOTS }
+      })
+    })
+  }
+
+  // don't show hidden plots
+  const isPlotHiddenInGrid = (plotId: PlotIdentifier) => {
+    const itemHidden = plotUiStateArray.find(
+      item =>
+        plotMatchesDefinition(plotId, item.plotId) && item.isHidden === true,
+    )
+    const result = itemHidden !== undefined
+    return result
+  }
+
+  const showMoreButtonState = useMemo<ShowMoreState>(() => {
+    if (
+      // if at least one item is hidden
+      plotUiStateArray.find(item => item.isHidden === true)
+    ) {
+      return 'MORE'
+    } else if (plotUiStateArray.length <= DEFAULT_VISIBLE_PLOTS) {
+      return 'NONE'
+    } else {
+      return 'LESS'
+    }
+  }, [plotUiStateArray])
+
+  // hides facet graph
+  const hidePlotInGrid = (plotId: PlotIdentifier) => {
+    setUiPropertyForPlot(plotId, 'isHidden', true)
+  }
+
+  const setPlotType = (plotId: PlotIdentifier, plotType: PlotType) => {
+    setUiPropertyForPlot(plotId, 'plotType', plotType)
+  }
+
+  const setUiPropertyForPlot = (
+    plotId: PlotIdentifier,
+    propName: keyof UiPlotState,
+    value: boolean | PlotType, // 'the possible values of the above type' (currently can't be specified in TS using symbols)
+  ) => {
+    setPlotUiStateArray(plotUiStateArray =>
+      plotUiStateArray.map(item =>
+        plotMatchesDefinition(plotId, item.plotId)
+          ? { ...item, [propName]: value }
+          : item,
+      ),
+    )
+  }
+
+  const colorTracker: { facet: UniqueFacetIdentifier; colorIndex: number }[] =
+    // Todo: fix to work with customPlots
+    getFacets(data, facetsToPlot).map((el, index) => {
+      return {
+        facet: { columnName: el.columnName, jsonPath: el.jsonPath },
+        colorIndex: index,
+      }
+    })
+
+  const generatePlotKey = (plotUiState: UiPlotState) => {
+    if ('__custom' in plotUiState.plotId) {
+      // For custom plots
+      return `custom-${plotUiState.plotId.title}`
+    } else {
+      // For facet plots
+      return `facet-${plotUiState.plotId.columnName}-${plotUiState.plotId.jsonPath}`
+    }
+  }
+
+  const hasFacetsOrFilters =
+    (lastQueryRequest?.query.selectedFacets !== undefined &&
+      lastQueryRequest.query.selectedFacets.length > 0) ||
+    (lastQueryRequest?.query.additionalFilters !== undefined &&
+      lastQueryRequest?.query.additionalFilters.length > 0)
+  if (error || (!data && isLoadingNewBundle)) {
+    return <></>
+  } else {
+    return (
+      <>
+        <TotalQueryResults
+          frontText={''}
+          endText={hasFacetsOrFilters ? 'filtered by' : ''}
+          hideIfUnfiltered={true}
+        />
```

Comment:
I think this should be pullled out of this component and moved to `QueryWrapperPlotNav`, but we need to make sure that this change doesn't affect other components